From 1ab762d96c1b036f0e947267141868e349a548da Mon Sep 17 00:00:00 2001
From: Isteevan Shetoo <isteevan.shetoo@is-info.fr>
Date: Mon, 26 Oct 2020 22:05:28 +0100
Subject: [PATCH] Override config path

---
 src/findlib/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/findlib/Makefile b/src/findlib/Makefile
index 6ec8a71..460dd38 100644
--- a/src/findlib/Makefile
+++ b/src/findlib/Makefile
@@ -88,7 +88,7 @@ findlib_config.ml: findlib_config.mlp $(TOP)/Makefile.config
 	USE_CYGPATH="$(USE_CYGPATH)"; \
 	export USE_CYGPATH; \
 	cat findlib_config.mlp | \
-	        $(SH) $(TOP)/tools/patch '@CONFIGFILE@' '$(OCAMLFIND_CONF)' | \
+	        $(SH) $(TOP)/tools/patch '@CONFIGFILE@' '$(OCAMLFIND_CONF_OVERRIDE)' | \
 	        $(SH) $(TOP)/tools/patch '@STDLIB@' '$(OCAML_CORE_STDLIB)' | \
 		sed -e 's;@AUTOLINK@;$(OCAML_AUTOLINK);g' \
 		    -e 's;@SYSTEM@;$(SYSTEM);g' \
-- 
2.29.1

